src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs

```diff
@@ -0,0 +1,172 @@
+                    }
+                    else if (!Enum.TryParse(command.Values[0], ignoreCase: true, result: out action))
+                    {
+                        throw new CommandLineArgumentCombinationException(string.Format(CultureInfo.CurrentCulture, Strings.Error_UnknownAction, command.Values[0]));
+                    }
+
+                    string name = null;
+
+                    if (command.Values.Count > 1)
+                    {
+                        name = command.Values[1];
+                    }
+
+                    string packagePath = null;
+                    string sourceUrl = null;
+                    string fingerPrint = null;
+                    if (command.Values.Count() > 2)
+                    {
+                        if (action == TrustCommand.Author || action == TrustCommand.Repository)
+                        {
+                            packagePath = command.Values[2];
+                        }
+                        else if (action == TrustCommand.Source)
+                        {
+                            sourceUrl = command.Values[2];
+                        }
+                        else if (action == TrustCommand.Certificate)
+                        {
+                            fingerPrint = command.Values[2];
+                        }
+                    }
+
+                    ISettings settings = ProcessConfigFile(configfile.Value());
+
+                    var trustedSignersArgs = new TrustedSignersArgs()
+                    {
+                        Action = MapTrustEnumAction(action),
+                        PackagePath = packagePath,
+                        Name = name,
+                        ServiceIndex = sourceUrl,
+                        CertificateFingerprint = fingerPrint,
+                        FingerprintAlgorithm = algorithm.Value(),
+                        AllowUntrustedRoot = allowuntrustedrootOption.HasValue(),
+                        Author = action == TrustCommand.Author,
+                        Repository = action == TrustCommand.Repository,
+                        Owners = CommandLineUtility.SplitAndJoinAcrossMultipleValues(owners.Values),
+                        Logger = getLogger()
+                    };
+
+                    setLogLevel(XPlatUtility.MSBuildVerbosityToNuGetLogLevel(verbosity.Value()));
+
+#pragma warning disable CS0618 // Type or member is obsolete
+                    var sourceProvider = new PackageSourceProvider(settings, enablePackageSourcesChangedEvent: false);
+#pragma warning restore CS0618 // Type or member is obsolete
+                    var trustedSignersProvider = new TrustedSignersProvider(settings);
+
+                    var runner = new TrustedSignersCommandRunner(trustedSignersProvider, sourceProvider);
+                    var trustedSignTask = runner.ExecuteCommandAsync(trustedSignersArgs);
+                    await trustedSignTask;
+
+                    return trustedSignTask.Result;
```

Comment:
Thank you for pointing out this problem, this can cause unwanted problem.   Addressed.
@kartheekp-ms You need to address below one for `verify` command too.
https://github.com/NuGet/NuGet.Client/blob/5138176d007034f96afce7132f034403a3888c44/src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/VerifyCommand.cs#L62-L64